osu.Game/Graphics/UserInterface/DialogButton.cs

```diff
@@ -79,6 +79,67 @@ public float TextSize
             }
         }
 
+        private bool isKeyboardSelected;
+        public bool IsKeyboardSelected
+        {
+            get
+            {
+                return isKeyboardSelected;
+            }
+            set
+            {
+                isKeyboardSelected = value;
+                if (value)
+                {
+                    spriteText.TransformSpacingTo(hoverSpacing, hover_duration, Easing.OutElastic);
+                    colourContainer.ResizeTo(new Vector2(hover_width, 1f), hover_duration, Easing.OutElastic);
+                    glowContainer.FadeIn(glow_fade_duration, Easing.Out);
+                }
+                else
+                {
+                    colourContainer.ResizeTo(new Vector2(0.8f, 1f), hover_duration, Easing.OutElastic);
+                    spriteText.TransformSpacingTo(Vector2.Zero, hover_duration, Easing.OutElastic);
+                    glowContainer.FadeOut(glow_fade_duration, Easing.Out);
+                }
+
+            }
+        }
+
+        private bool isMouseSelected;
+        public bool IsMouseSelected
+        {
+            get
+            {
+                return isMouseSelected;
+            }
+            set
+            {
+                isMouseSelected = value;
+                if (value)
+                {
+                    spriteText.TransformSpacingTo(hoverSpacing, hover_duration, Easing.OutElastic);
+                    colourContainer.ResizeTo(new Vector2(hover_width, 1f), hover_duration, Easing.OutElastic);
+                    glowContainer.FadeIn(glow_fade_duration, Easing.Out);
+                }
+                else
+                {
+                    colourContainer.ResizeTo(new Vector2(0.8f, 1f), hover_duration, Easing.OutElastic);
+                    spriteText.TransformSpacingTo(Vector2.Zero, hover_duration, Easing.OutElastic);
+                    glowContainer.FadeOut(glow_fade_duration, Easing.Out);
+                }

```

Comment:
How about a separate `updateSelectedState()` method that does these transformations based on the value of `IsSelected`?